sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1628,6 +1642,46 @@ static bool setEnvVar(const char *name, const char *value) {
   return true;
 }
 
+static class ZeUSMImport {
```

Comment:
please do the extension lookup similar to how we do several others, see https://github.com/intel/llvm/blob/537336232d28fde30b068040b356664c727f583d/sycl/plugins/level_zero/pi_level_zero.cpp#L1661

Reply:
There is existing usage of **zeDriverGetExtensionProperties**. However, the Import/Release extension is done through a new API, **zeDriverGetExtensionFunctionAddress**, which has no existing usage in the plugin. I suggest that the mechanism added here for using this API for Import/Release can later be extended for other new function adddress, as they are implemented by L0.

Reply:
Why is this extension is not reported by zeDriverGetExtensionProperties? Why do you need to call new API by address instead of by name as done currently by other extensions?

Reply:
zeDriverGetExtensionProperties allows querying for properties. It returns an array of properties, which are a pair: char name[ZE_MAX_EXTENSION_NAME] and uint32_t version. The names are predefined and the version value tells which version is supported.

zeDriverGetExtensionFunctionAddress is entirely different. It allows querying whether an L0 API name is available to be called.  A call to zeDriverGetExtensionFunctionAddress  with argument say, "zeXYZ", returns the address of the function zeXYZ within the driver if it is defined, else it returns 0. When a non-zero is returned it means the function zeXYZ can be invoked when needed. We use this functionality to check if "zexDriverImportExternalPointer" and "zexDriverReleaseImportedPointer" are available to be called. 

Thus, these two types of extensions are entirely different. The existing property mechanism tells us the version level of some predefined L0 features. The new API tells us if certain new L0 APIs are available to be called.

Reply:
> The names are predefined and the version value tells which version is supported.

The name/version should tell you which extension API names are there.

> Thus, these two types of extensions are entirely different. 

I don't see why there should be different types of using extensions.

Reply:
Based on offline discussion, this is an "experimental" L0 API and thus it is not reported by the "standard" extension query zeDriverGetExtensionProperties. Please add a comment explaining why this extension query is different.